cms/djangoapps/contentstore/views/tests/test_organizations.py

```diff
@@ -32,3 +41,62 @@ def test_organization_list(self):
         self.assertEqual(response.status_code, 200)
         org_names = json.loads(response.content)
         self.assertEqual(org_names, self.org_short_names)
+
+
+@patch.dict('django.conf.settings.FEATURES', {'ORGANIZATIONS_APP': True})
+@override_switch(settings.ENABLE_EDLY_ORGANIZATIONS_SWITCH, active=True)
+class TestEdlyOrganizationListing(TestCase):
+    """
+    Verify Organization listing behavior.
+    """
+    @patch.dict('django.conf.settings.FEATURES', {'ORGANIZATIONS_APP': True})
+    def setUp(self):
+        super(TestEdlyOrganizationListing, self).setUp()
+        self.staff = UserFactory(is_staff=True)
+        self.client.login(username=self.staff.username, password='test')
+        self.org_names_listing_url = reverse('organizations')
+
+    def test_without_authentication(self):
+        """
+        Verify authentication is required when accessing the endpoint.
+        """
+        self.client.logout()
+        response = self.client.get(self.org_names_listing_url)
+        assert response.status_code == 302
+
+    def test_organization_list(self):
+        """
+        Verify that the organization names list API only returns Edly's enabled organizations.
+        """
+
+        studio_site = SiteFactory()
+        edly_organization = EdlyOrganizationFactory(name='Test Edly Organization Name')
+        edly_sub_organization = EdlySubOrganizationFactory(
+            studio_site=studio_site,
+            edly_organization=edly_organization
+        )
+
+        response = self.client.get(self.org_names_listing_url, HTTP_ACCEPT='application/json', SERVER_NAME=studio_site.domain)
+
+        assert response.status_code == 200
+        assert len(response.json()) == 1
+        assert response.json()[0] == edly_sub_organization.edx_organization.short_name
+
+    def test_organization_list_without_linked_edly_sub_organization(self):
+        """
+        Verify that if there is no "EdlySubOrganization" linked to a studio site the organization names list API returns empty response.
+        """
+        studio_site = SiteFactory()
+        with LogCapture(LOGGER_NAME) as logger:
+            response = self.client.get(self.org_names_listing_url, HTTP_ACCEPT='application/json', SERVER_NAME=studio_site.domain)
+
+            logger.check(
+                (
+                    LOGGER_NAME,
+                    'ERROR',
```

Comment:
`EXCEPTION`

Reply:
It returns ERROR type

![image](https://user-images.githubusercontent.com/7139602/81392393-08335780-9138-11ea-8542-7f1aaa33ca6f.png)

Reply:
Lets go with `ERROR` type for now.